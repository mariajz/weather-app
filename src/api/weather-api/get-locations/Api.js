import ApiService from '../../ApiService';
import baseUrl from '../BaseUrl';

class Api extends ApiService {
    constructor({ queryParams }) {
        super({
            method: 'GET',
            url: `${baseUrl}/v1/search.json`,
            queryParams: queryParams,
        });
    }
}

export default Api;
