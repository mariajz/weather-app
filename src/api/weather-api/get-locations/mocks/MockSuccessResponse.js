export const mockSuccessResponse = [
    {
        id: 1120673,
        name: 'Kochi',
        region: 'Kerala',
        country: 'India',
        lat: 9.97,
        lon: 76.23,
        url: 'kochi-kerala-india',
    },
    {
        id: 3125114,
        name: 'Kochi-Shi',
        region: 'Ko¬çchi',
        country: 'Japan',
        lat: 33.56,
        lon: 133.53,
        url: 'kochi-shi-kochi-japan',
    },
    {
        id: 1120674,
        name: 'Kochinda',
        region: 'Orissa',
        country: 'India',
        lat: 21.73,
        lon: 84.35,
        url: 'kochinda-orissa-india',
    },
];
