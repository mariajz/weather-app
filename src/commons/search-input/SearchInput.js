import React from 'react';
import { TextInput } from 'react-native';
import { SearchInputWrapper } from './SearchInput.style';
import { Icon } from '../visual-elements';

const SearchInput = ({ placeholder }) => {
    return (
        <SearchInputWrapper>
            <TextInput
                placeholder={placeholder}
                placeholderTextColor={'black'}
            />
            <Icon iconName="search" />
        </SearchInputWrapper>
    );
};

export default SearchInput;
