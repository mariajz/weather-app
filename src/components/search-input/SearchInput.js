import React, { useState } from 'react';
import { View } from 'react-native';
import { Divider } from '../../commons/styles';
import { CustomIcon } from '../../commons/visual-elements';
import {
    DropDown,
    ModalWrapper,
    Row,
    SearchInputWrapper,
    StyledCustomIcon,
    StyledIcon,
    StyledModal,
    StyledText,
    StyledTextInput,
    StyledTouchableOpacity,
} from './SearchInput.style';

const SearchInput = ({ placeholder }) => {
    const [showDropDown, setShowDropDown] = useState(false);
    const [showModal, setShowModal] = useState(true);

    const [searchIconPressed, setSearchIconPressed] = useState(false);

    const [locations] = useState([
        { name: 'abc', country: 'India', key: 1 },
        { name: 'abc', country: 'India', key: 2 },
        { name: 'abc', country: 'India', key: 3 },
        { name: 'abc', country: 'India', key: 4 },
        { name: 'abc', country: 'India', key: 5 },
    ]);

    const handleOnSearchIconPress = () => {
        setSearchIconPressed(!searchIconPressed);
        setShowModal(true);
    };

    const handleOnChangeText = data => {
        if (data.length === 0) {
            setShowDropDown(false);
        }
        if (data.length > 3) {
            // make api call to fetch locations
            if (locations.length !== 0) {
                setShowDropDown(true);
            } else {
                setShowDropDown(false);
            }
        }
    };

    const handleOnDropDownItemPress = () => {
        // make api call to fetch forecast data
    };

    return (
        <SearchInputWrapper>
            <If condition={searchIconPressed}>
                <ModalWrapper>
                    <StyledTouchableOpacity
                        onPress={() => {
                            setShowModal(true);
                        }}>
                        <View>
                            <StyledModal
                                visible={showModal}
                                onBackdropPress={() => {
                                    setShowDropDown(false);
                                    setShowModal(false);
                                    setSearchIconPressed(!searchIconPressed);
                                }}>
                                <StyledTextInput
                                    placeholder={placeholder}
                                    placeholderTextColor="black"
                                    onChangeText={handleOnChangeText}
                                />
                                <If condition={showDropDown}>
                                    <DropDown>
                                        <If condition={locations.length > 0}>
                                            {locations.map(
                                                (location, index) => {
                                                    return (
                                                        <React.Fragment
                                                            key={index}>
                                                            <Row
                                                                onPress={() => {
                                                                    handleOnDropDownItemPress(
                                                                        location,
                                                                    );
                                                                }}>
                                                                <CustomIcon
                                                                    iconName="pin"
                                                                    size={12}
                                                                    tintColor="black"
                                                                />
                                                                <StyledText>
                                                                    {
                                                                        location?.name
                                                                    }
                                                                    ,
                                                                    {
                                                                        location?.country
                                                                    }
                                                                </StyledText>
                                                            </Row>
                                                            <Divider />
                                                        </React.Fragment>
                                                    );
                                                },
                                            )}
                                        </If>
                                    </DropDown>
                                </If>
                            </StyledModal>
                        </View>
                    </StyledTouchableOpacity>
                </ModalWrapper>
            </If>
            <StyledIcon>
                <If condition={searchIconPressed}>
                    <StyledCustomIcon
                        iconName="close"
                        onPress={handleOnSearchIconPress}
                        tintColor="black"
                        size="14"
                    />
                    <Else />
                    <CustomIcon
                        testID="search-icon"
                        iconName="search"
                        onPress={handleOnSearchIconPress}
                        tintColor="black"
                    />
                </If>
            </StyledIcon>
        </SearchInputWrapper>
    );
};

export default SearchInput;
