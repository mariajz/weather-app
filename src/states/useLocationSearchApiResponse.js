import { atom, useAtom } from 'jotai';

const locationSearchApiResponseAtom = atom(undefined);

const useLocationSearchApiResponse = () => {
    const [response, setResponse] = useAtom(locationSearchApiResponseAtom);

    return {
        response,
        setResponse,
    };
};
export default useLocationSearchApiResponse;
