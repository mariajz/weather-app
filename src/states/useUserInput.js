import { atom, useAtom } from 'jotai';

const userInputAtom = atom(undefined);

const useUserInput = () => {
    const [userInput, setUserInput] = useAtom(userInputAtom);

    return {
        userInput,
        setUserInput,
    };
};
export default useUserInput;
