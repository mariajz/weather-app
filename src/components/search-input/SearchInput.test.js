import { fireEvent, render, act } from '@testing-library/react-native';
import React from 'react';
import { View as MockView } from 'react-native';
import SearchInput from './SearchInput';
import useLocationSearchApiResponse from '../../states/useLocationSearchApiResponse';
import { mockSuccessResponse } from '../../api/weather-api/get-locations/mocks';
import useUserInput from '../../states/useUserInput';

jest.mock('../../commons/visual-elements', () => ({
    CustomIcon: props => <MockView {...props}>Search Icon</MockView>,
    Theme: {
        BackgroundWhite: jest.fn(
            opacity => `mocked rgba(255, 255, 255, ${opacity})`,
        ),
    },
}));

jest.mock('../../states/useLocationSearchApiResponse');

const mockSetSearchLocation = jest.fn();
jest.mock('../../states/useSearchLocation', () => () => ({
    setSearchLocation: mockSetSearchLocation,
}));

const mockSetUserInput = jest.fn();
jest.mock('../../states/useUserInput');

jest.mock('../../commons/styles', () => ({
    Divider: () => <MockView />,
}));

const mockHandleFetchLocationData = jest.fn();
jest.mock('../../hooks/useGetAllLocations', () => () => ({
    handleFetchLocationData: mockHandleFetchLocationData,
}));

const mockHandleFetchWeather = jest.fn();
jest.mock('../../hooks/useGetCurrentWeather', () => () => ({
    handleFetchWeather: mockHandleFetchWeather,
}));

jest.mock('react-native-modal', () => ({ children, ...restProps }) => (
    <MockView {...restProps}>{children}</MockView>
));

jest.mock('../search-city-section/DetailRow', () => props => (
    <MockView {...props} />
));

const renderDropdown = async () => {
    jest.useFakeTimers();

    const container = render(<SearchInput />);
    const { getByTestId, queryByTestId, debug } = container;

    await act(async () => {
        getByTestId('search-icon').props.onPress();
    });

    await act(async () => {
        fireEvent.press(getByTestId('text-input-wrapper'));
    });

    await act(async () => {
        getByTestId('text-input').props.onChangeText('abcd');
    });

    act(() => {
        jest.advanceTimersByTime(2000);
    });

    return { container, getByTestId, queryByTestId, debug };
};

const mockSetError = jest.fn();

describe('SearchInput', () => {
    beforeEach(() => {
        useLocationSearchApiResponse.mockImplementation(() => ({
            response: undefined,
            error: undefined,
            setError: mockSetError,
        }));
        useUserInput.mockImplementation(() => ({
            userInput: 'Delhi',
            setUserInput: mockSetUserInput,
        }));
    });
    afterEach(() => {
        jest.clearAllMocks();
    });

    it('should render SearchInput', () => {
        const container = render(<SearchInput />);

        expect(container).toMatchSnapshot();
    });

    it('should open text input on search icon press', async () => {
        const container = render(<SearchInput />);

        const { getByTestId } = container;

        getByTestId('search-icon').props.onPress();

        expect(getByTestId('text-input-wrapper')).toBeDefined();
        expect(getByTestId('text-input')).toBeDefined();
        expect(container).toMatchSnapshot();
    });

    it('should open the modal on clicking on text input and close the modal and text input on backdrop press', () => {
        const container = render(<SearchInput />);

        const { getByTestId, queryByTestId } = container;

        getByTestId('search-icon').props.onPress();

        expect(getByTestId('text-input-wrapper')).toBeDefined();
        expect(getByTestId('text-input')).toBeDefined();

        fireEvent.press(getByTestId('text-input-wrapper'));

        expect(getByTestId('modal')).toBeDefined();

        getByTestId('modal').props.onBackdropPress();

        expect(queryByTestId('text-input-wrapper')).toBeNull();
        expect(queryByTestId('modal')).toBeNull();
    });

    describe('Tests for dropdown', () => {
        beforeEach(() => {
            useLocationSearchApiResponse.mockImplementation(() => ({
                response: mockSuccessResponse,
                error: false,
                setError: mockSetError,
            }));
        });
        afterEach(() => {
            jest.useRealTimers();
            jest.clearAllMocks();
        });

        it('should not show dropdown when entered input text length <3', async () => {
            jest.useFakeTimers();
            useUserInput.mockImplementation(() => ({
                userInput: undefined,
                setUserInput: mockSetUserInput,
            }));

            useLocationSearchApiResponse.mockImplementation(() => ({
                response: undefined,
                error: undefined,
                setError: mockSetError,
            }));
            const container = render(<SearchInput />);
            const { getByTestId, queryByTestId } = container;

            await act(async () => {
                getByTestId('search-icon').props.onPress();
            });

            await act(async () => {
                fireEvent.press(getByTestId('text-input-wrapper'));
            });

            await act(async () => {
                getByTestId('text-input').props.onChangeText('ab');
            });

            act(() => {
                jest.advanceTimersByTime(2000);
            });

            expect(getByTestId('text-input').props.value).toStrictEqual('ab');
            expect(mockHandleFetchLocationData).toHaveBeenCalledTimes(0);
            expect(queryByTestId('dropdown')).toBeNull();
        });

        it('should show dropdown when entered input text length >3 and collapse the dropdown when length is 0', async () => {
            const { getByTestId, queryByTestId } = await renderDropdown();

            expect(getByTestId('dropdown')).toBeDefined();
            expect(mockSetUserInput).toHaveBeenCalledTimes(1);
            expect(mockHandleFetchLocationData).toHaveBeenCalledTimes(1);

            await act(async () => {
                getByTestId('text-input').props.onChangeText('');
            });

            useLocationSearchApiResponse.mockImplementation(() => ({
                response: undefined,
                error: undefined,
                setError: mockSetError,
            }));

            act(() => {
                jest.advanceTimersByTime(2000);
            });

            expect(getByTestId('text-input').props.value).toStrictEqual('');
            expect(queryByTestId('dropdown')).toBeNull();
        });

        it('should populate the response in dropdown when response is not undefined', async () => {
            const { container, getByTestId } = await renderDropdown();

            expect(getByTestId('dropdown')).toBeDefined();

            expect(container).toMatchSnapshot();
        });

        it('should not populate the response in dropdown and show unavailable message when response is empty', async () => {
            useLocationSearchApiResponse.mockImplementation(() => ({
                response: [],
                error: undefined,
                setError: mockSetError,
            }));
            const { container, getByTestId } = await renderDropdown();

            expect(getByTestId('dropdown')).toBeDefined();
            expect(getByTestId('item-unavailable')).toBeDefined();

            expect(container).toMatchSnapshot();
        });

        it('should collapse the dropdown and text input when error is true and popup is shown', async () => {
            useLocationSearchApiResponse.mockImplementation(() => ({
                response: undefined,
                error: true,
                setError: mockSetError,
            }));

            const container = render(<SearchInput />);

            expect(container).toMatchSnapshot();
        });

        it('should set latitude and longitude to searchLocation and call ForecastApi on pressing dropdown item', async () => {
            const { getByTestId } = await renderDropdown();

            expect(getByTestId('dropdown')).toBeDefined();

            await act(async () => {
                getByTestId('item-0').props.handleOnDropDownItemPress();
            });

            expect(mockSetSearchLocation).toHaveBeenCalledTimes(1);
            expect(mockSetSearchLocation).toHaveBeenCalledWith('9.97,76.23');
            expect(mockHandleFetchWeather).toHaveBeenCalledTimes(1);
        });
    });
});
