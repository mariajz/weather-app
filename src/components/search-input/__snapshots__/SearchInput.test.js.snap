// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchInput Tests for dropdown should not populate the response in dropdown and show unavailable message when response is empty 1`] = `
<View
  style={
    {
      "height": "8%",
    }
  }
>
  <View
    style={
      {
        "backgroundColor": "mocked rgba(255, 255, 255, 0.2)",
        "borderBottomLeftRadius": 20,
        "borderBottomRightRadius": 20,
        "borderTopLeftRadius": 20,
        "borderTopRightRadius": 20,
        "height": 40,
        "marginBottom": 10,
        "marginLeft": 10,
        "marginRight": 10,
        "marginTop": 10,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "height": "100%",
          "opacity": 1,
          "width": "100%",
        }
      }
      testID="text-input-wrapper"
    >
      <View>
        <View
          onBackdropPress={[Function]}
          style={
            {
              "position": "absolute",
              "top": 48,
              "width": "85%",
            }
          }
          testID="modal"
          visible={true}
        >
          <TextInput
            onChangeText={[Function]}
            placeholderTextColor="black"
            style={
              {
                "paddingLeft": 10,
              }
            }
            testID="text-input"
            value="abcd"
          />
          <View
            style={
              {
                "backgroundColor": "white",
                "borderBottomLeftRadius": 10,
                "borderBottomRightRadius": 10,
                "borderTopLeftRadius": 10,
                "borderTopRightRadius": 10,
                "marginTop": 15,
              }
            }
            testID="dropdown"
          >
            <View
              locationAvailable={false}
              testID="item-unavailable"
            />
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      {
        "backgroundColor": "mocked rgba(255, 255, 255, 0.5)",
        "borderBottomLeftRadius": 20,
        "borderBottomRightRadius": 20,
        "borderTopLeftRadius": 20,
        "borderTopRightRadius": 20,
        "height": 40,
        "marginTop": 10,
        "paddingBottom": 7,
        "paddingLeft": 7,
        "paddingRight": 7,
        "paddingTop": 7,
        "position": "absolute",
        "right": 12,
        "width": 40,
      }
    }
  >
    <View
      iconName="close"
      onPress={[Function]}
      size="14"
      style={
        {
          "marginBottom": 6,
          "marginLeft": 6,
          "marginRight": 6,
          "marginTop": 6,
        }
      }
      tintColor="black"
    >
      Search Icon
    </View>
  </View>
</View>
`;

exports[`SearchInput Tests for dropdown should populate the response in dropdown when response is not undefined 1`] = `
<View
  style={
    {
      "height": "8%",
    }
  }
>
  <View
    style={
      {
        "backgroundColor": "mocked rgba(255, 255, 255, 0.2)",
        "borderBottomLeftRadius": 20,
        "borderBottomRightRadius": 20,
        "borderTopLeftRadius": 20,
        "borderTopRightRadius": 20,
        "height": 40,
        "marginBottom": 10,
        "marginLeft": 10,
        "marginRight": 10,
        "marginTop": 10,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "height": "100%",
          "opacity": 1,
          "width": "100%",
        }
      }
      testID="text-input-wrapper"
    >
      <View>
        <View
          onBackdropPress={[Function]}
          style={
            {
              "position": "absolute",
              "top": 48,
              "width": "85%",
            }
          }
          testID="modal"
          visible={true}
        >
          <TextInput
            onChangeText={[Function]}
            placeholderTextColor="black"
            style={
              {
                "paddingLeft": 10,
              }
            }
            testID="text-input"
            value="abcd"
          />
          <View
            style={
              {
                "backgroundColor": "white",
                "borderBottomLeftRadius": 10,
                "borderBottomRightRadius": 10,
                "borderTopLeftRadius": 10,
                "borderTopRightRadius": 10,
                "marginTop": 15,
              }
            }
            testID="dropdown"
          >
            <View
              handleOnDropDownItemPress={[Function]}
              index={0}
              location={
                {
                  "country": "India",
                  "id": 1120673,
                  "lat": 9.97,
                  "lon": 76.23,
                  "name": "Kochi",
                  "region": "Kerala",
                  "url": "kochi-kerala-india",
                }
              }
              locationAvailable={true}
              testID="item-0"
            />
            <View
              handleOnDropDownItemPress={[Function]}
              index={1}
              location={
                {
                  "country": "Japan",
                  "id": 3125114,
                  "lat": 33.56,
                  "lon": 133.53,
                  "name": "Kochi-Shi",
                  "region": "Kochi",
                  "url": "kochi-shi-kochi-japan",
                }
              }
              locationAvailable={true}
              testID="item-1"
            />
            <View
              handleOnDropDownItemPress={[Function]}
              index={2}
              location={
                {
                  "country": "India",
                  "id": 1120674,
                  "lat": 21.73,
                  "lon": 84.35,
                  "name": "Kochinda",
                  "region": "Orissa",
                  "url": "kochinda-orissa-india",
                }
              }
              locationAvailable={true}
              testID="item-2"
            />
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      {
        "backgroundColor": "mocked rgba(255, 255, 255, 0.5)",
        "borderBottomLeftRadius": 20,
        "borderBottomRightRadius": 20,
        "borderTopLeftRadius": 20,
        "borderTopRightRadius": 20,
        "height": 40,
        "marginTop": 10,
        "paddingBottom": 7,
        "paddingLeft": 7,
        "paddingRight": 7,
        "paddingTop": 7,
        "position": "absolute",
        "right": 12,
        "width": 40,
      }
    }
  >
    <View
      iconName="close"
      onPress={[Function]}
      size="14"
      style={
        {
          "marginBottom": 6,
          "marginLeft": 6,
          "marginRight": 6,
          "marginTop": 6,
        }
      }
      tintColor="black"
    >
      Search Icon
    </View>
  </View>
</View>
`;

exports[`SearchInput should open text input on search icon press 1`] = `
<View
  style={
    {
      "height": "8%",
    }
  }
>
  <View
    style={
      {
        "backgroundColor": "mocked rgba(255, 255, 255, 0.2)",
        "borderBottomLeftRadius": 20,
        "borderBottomRightRadius": 20,
        "borderTopLeftRadius": 20,
        "borderTopRightRadius": 20,
        "height": 40,
        "marginBottom": 10,
        "marginLeft": 10,
        "marginRight": 10,
        "marginTop": 10,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "height": "100%",
          "opacity": 1,
          "width": "100%",
        }
      }
      testID="text-input-wrapper"
    >
      <View>
        <View
          onBackdropPress={[Function]}
          style={
            {
              "position": "absolute",
              "top": 48,
              "width": "85%",
            }
          }
          testID="modal"
          visible={true}
        >
          <TextInput
            onChangeText={[Function]}
            placeholderTextColor="black"
            style={
              {
                "paddingLeft": 10,
              }
            }
            testID="text-input"
            value=""
          />
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      {
        "backgroundColor": "mocked rgba(255, 255, 255, 0.5)",
        "borderBottomLeftRadius": 20,
        "borderBottomRightRadius": 20,
        "borderTopLeftRadius": 20,
        "borderTopRightRadius": 20,
        "height": 40,
        "marginTop": 10,
        "paddingBottom": 7,
        "paddingLeft": 7,
        "paddingRight": 7,
        "paddingTop": 7,
        "position": "absolute",
        "right": 12,
        "width": 40,
      }
    }
  >
    <View
      iconName="close"
      onPress={[Function]}
      size="14"
      style={
        {
          "marginBottom": 6,
          "marginLeft": 6,
          "marginRight": 6,
          "marginTop": 6,
        }
      }
      tintColor="black"
    >
      Search Icon
    </View>
  </View>
</View>
`;

exports[`SearchInput should render SearchInput 1`] = `
<View
  style={
    {
      "height": "8%",
    }
  }
>
  <View
    style={
      {
        "backgroundColor": "mocked rgba(255, 255, 255, 0.5)",
        "borderBottomLeftRadius": 20,
        "borderBottomRightRadius": 20,
        "borderTopLeftRadius": 20,
        "borderTopRightRadius": 20,
        "height": 40,
        "marginTop": 10,
        "paddingBottom": 7,
        "paddingLeft": 7,
        "paddingRight": 7,
        "paddingTop": 7,
        "position": "absolute",
        "right": 12,
        "width": 40,
      }
    }
  >
    <View
      iconName="search"
      onPress={[Function]}
      testID="search-icon"
      tintColor="black"
    >
      Search Icon
    </View>
  </View>
</View>
`;
